### PR TITLE
migrate openapi to typescript

### PR DESCRIPTION
**Pull request by atakanyenel**

### CODE REVIEW

File: openapi-mock/Dockerfile [hunk 1]
```diff
@@ -1,12 +1,20 @@
-FROM node:11-alpine
+FROM node:11-alpine as builder
 RUN apk --no-cache upgrade
```
Comment (a-thaler) [inline openapi-mock/Dockerfile]: The update of the packages mainly has to happen in the productive image, not the build image. To assure known security vulnerabilities in the used packages are fixed.

File: openapi-mock/Dockerfile [hunk 2]
```diff
@@ -1,12 +1,20 @@
...
+COPY --from=builder /app/dist /app/dist
+WORKDIR /app
+
 COPY package*.json ./
 
 ENV NODE_ENV=production
 RUN npm install
-COPY . .
 
-EXPOSE 10000
```
Comment (a-thaler) [inline openapi-mock/Dockerfile]: Why you do not expose the port anymore? How to access the application?

File: openapi-mock/package.json [hunk 3]
```diff
@@ -16,22 +16,31 @@
...
-    "winston": "^3.2.1"
+    "winston": "^3.2.1",
+    "js-yaml": "^3.13.0"
   },
   "devDependencies": {
+    "@types/express": "^4.16.1",
+    "@types/js-yaml": "^3.12.0",
+    "@types/mocha": "^5.2.6",
+    "@types/morgan": "^1.7.35",
+    "@types/supertest": "^2.0.7",
```
Comment (a-thaler) [inline openapi-mock/package.json]: Why you have supertest as a dependency with and without types and in different versions?
Why there is a dependency to the swagger-express-middleware? In an old version

Comment (atakanyenel) [reply openapi-mock/package.json]: `@types` are declaration libraries used by typescript's Intellisense. A package can choose to publish it's own declarations with the package itself (like us) or they can publish their declarations in a separate package to `@types` scope. Those packages only include declarations and no code. It is not mandatory to publish them every time with the real package so they might have a different version.

Comment (a-thaler) [reply openapi-mock/package.json]: Thanks for the explanation, well done :)

**Pull request merged**